smoke-tests/apps/PreAggMetricsWithRoleNameOverridesAndSampling/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PreAggMetricsWithRoleNameOverridesAndSamplingTest.java

```diff
@@ -66,9 +66,11 @@ private void testApp(String roleName) throws Exception {
     for (Envelope envelope : metricsEnvelops) {
       MetricData metricData = (MetricData) ((Data<?>) envelope.getData()).getBaseData();
       String name = metricData.getMetrics().get(0).getName();
-      if ("http.client.request.duration".equals(name)) {
+      if ("http.client.request.duration".equals(name)
+          && metricData.getProperties().get("_MS.MetricId") != null) {
```

Comment:
(since this is the important part)
```suggestion
      if ("dependencies/duration".equals(metricData.getProperties().get("_MS.MetricId"))) {
```